atc/api/containerserver/report.go

```diff
@@ -41,6 +42,24 @@ func (s *Server) ReportWorkerContainers(w http.ResponseWriter, r *http.Request)
 		"num-handles": len(handles),
 	})
 
+	numUnknownContainers, err := s.containerRepository.DestroyUnknownContainers(workerName, handles)
+	if err != nil {
+		logger.Error("failed-to-destroy-unknown-containers", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	if numUnknownContainers > 0 {
+		logger.Info("unknown-container-handles", lager.Data{
+			"worker-name":   workerName,
+			"handles-count": numUnknownContainers,
+		})
+
+		metric.WorkerUnknownContainers{
```

Comment:
as we represent `unknown containers` as a gauge on pull-based "emitters" (prometheus at the moment), we can get into the following:

- t0: `numUnknownContainers == 10`
  - gauge = 10
- t1: `numUnknownContainers == 0`
  - gauge = 10
- t2: `numUnknownContainers == 0`
  - gauge = 10
- t3: `numUnknownContainers == 0`
  - gauge = 10

It'd be better if the metric emission happened outside that condition 🤔

Reply:
good catch